### PR TITLE
Azure testing server

### PR DESCRIPTION
**Pull request by princebaretto99**

### CODE REVIEW

File: browserstack-examples-appium-nunit/Browserstack/App/Tests/ReportGeneration.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        [Test]
+        public async Task ReportGenerator()
+        {
+            var BuildId = "";
+            var Limit = 100;
+            var Offset = 0;
+            var userName = Environment.GetEnvironmentVariable("BROWSERSTACK_USERNAME") ?? "";
+            var accessKey = Environment.GetEnvironmentVariable("BROWSERSTACK_ACCESS_KEY") ?? "";
+            var buildName = Environment.GetEnvironmentVariable("BROWSERSTACK_BUILD_NAME") ?? "";
+            //var buildName = "Sales Demo";
```
Comment (samirans89) [inline browserstack-examples-appium-nunit/Browserstack/App/Tests/ReportGeneration.cs:24]: remove commented lines of code.

File: browserstack-examples-appium-nunit/Browserstack/App/Tests/ReportGeneration.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        "<!DOCTYPE html><html><head><style>table {  font-family: arial, sans-serif; border-collapse: collapse; width: 100%;}td, th{ border: 1px solid #dddddd;  text-align: left;  padding: 8px;}</style></head><body><table>",
+        "<tr><th>Build Name</th><th>Project Name</th><th>Session ID</th><th>Session Name</th><th>OS Name</th><th>OS Version</th><th>Device</th><th>Status</tr>"
+
+        };
+
+            string[] htmlEnd = { "</table></body></html>" };
+
+            await System.IO.File.WriteAllLinesAsync(fileLocation, htmlStart);
+
+            var client = new RestClient("https://api.browserstack.com");
```
Comment (samirans89) [inline browserstack-examples-appium-nunit/Browserstack/App/Tests/ReportGeneration.cs:39]: Make this part `https://api.browserstack.com` a variable and use it across the script

File: browserstack-examples-appium-nunit/Browserstack/App/Tests/ReportGeneration.cs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            client.Authenticator = new HttpBasicAuthenticator(userName, accessKey);
+
+            var buildApiRequest = new RestRequest("https://api.browserstack.com/app-automate/builds.json?limit=40");
```
Comment (samirans89) [inline browserstack-examples-appium-nunit/Browserstack/App/Tests/ReportGeneration.cs:42]: Refer this part `https://api.browserstack.com` from the variable you have defined earlier

**Review by samirans89 — CHANGES_REQUESTED** [review]

File: Reports/output.html [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<!DOCTYPE html><html><head><style>table {  font-family: arial, sans-serif; border-collapse: collapse; width: 100%;}td, th{ border: 1px solid #dddddd;  text-align: left;  padding: 8px;}</style></head><body><table>
```
Comment (samirans89) [inline Reports/output.html:1]: we don't need this file checked in.

File: browserstack-examples-appium-nunit/Browserstack/App/Tests/ReportGeneration.cs (hunk 1 above)
Comment (princebaretto99) [reply browserstack-examples-appium-nunit/Browserstack/App/Tests/ReportGeneration.cs]: Done

File: browserstack-examples-appium-nunit/Browserstack/App/Tests/ReportGeneration.cs (hunk 2 above)
Comment (princebaretto99) [reply browserstack-examples-appium-nunit/Browserstack/App/Tests/ReportGeneration.cs]: Done

File: browserstack-examples-appium-nunit/Browserstack/App/Tests/ReportGeneration.cs (hunk 3 above)
Comment (princebaretto99) [reply browserstack-examples-appium-nunit/Browserstack/App/Tests/ReportGeneration.cs]: Done

File: Reports/output.html (hunk 4 above)
Comment (princebaretto99) [reply Reports/output.html]: Done

**Review by samirans89 — APPROVED** [review]

**Pull request merged**